### ISSUE TITLE
keyDownAction triggers a warning/error (sound) message on Mac

### ISSUE THREAD

**Issue by Xon77** [supercollider/supercollider#4058]
Hi,

There seems to be other issues as #3878  related to keyDownAction, but on SC 3.10.0-beta2 with Mac 10.13.6, the code below triggers for me a warning/error (sound) message, when you press a key :

```supercollider
w = Window("my name is... panel", Rect(128, 64, 340, 360)); w.front;
w.view.keyDownAction_({ |view, char, mod, uni, key, key2|
key.postln;
});
```

**Comment by Xon77** [comment]
It is maybe related to this pull request #3972 ?

**Comment by brianlheim** [comment]
@Xon77 Thanks for the report. You can use triple backticks with "```supercollider" to get syntax highlighting. I have edited your report to add them. Also, please fill out the template as given next time, it makes it easier for us to understand the report.

Is this new behavior in 3.10.0-beta2? When was it last working correctly?

**Comment by jpburstrom** [comment]
@Xon77 I can reproduce this. A workaround (fix?) would be to return true from the function, like this:
```supercollider
w = Window("my name is... panel", Rect(128, 64, 340, 360)); w.front;
w.view.keyDownAction_({ |view, char, mod, uni, key, key2|
	key.postln;
	true
});
```
Also, View.defaultKeyDownAction doesn't return true, which causes the same behaviour on all views with undefined keyDownActions.

**Comment by Xon77** [comment]
This new behaviour is in 3.10.0-beta2. It was working correctly with 3.9.3.

**Comment by Xon77** [comment]
Thanks @jpburstrom. "true" solves the issue.
Do we have now to add "true" to keyDownAction to solve the warning sound message on Mac ?
Just want to know if this syntax with "true" is now the "standard" with SC 3.10.
Thanks

**Comment by jamshark70** [comment]
> Just want to know if this syntax with "true" is now the "standard" with SC 3.10.

No, it shouldn't be. 3.10 should not break this. (3.10's final release had better not break this.) That would be an awful backward compatibility fail otherwise.

It's a troubleshooting test. It means that trapping the keystroke and not allowing it to propagate up to the parent prevents the alert bell. That provides a clue to where the problem is happening.

**Comment by brianlheim** [comment]
Also good to know would be what keys people are pressing to test this, and if there is any key that does not have this behavior. But I think it's pretty clear at this point where to start looking.

**Comment by Xon77** [comment]
The bug seems to concern all keys, except modifiers.

**Comment by brianlheim** [comment]
Also note - this is most like something that came out of upgrading our version of Qt and associated changes, so I would take a look at #3165, #3960, and #3936 for anything suspicious.

**Comment by brianlheim** [comment]
#4105 may fix this; a test build will be available at http://supercollider.s3.amazonaws.com/builds/supercollider/supercollider/osx/issue/4058-latest.html soon. I would encourage anyone who has time to test this on a couple of window/event snippets, because I'm not sure what cases might reveal problems with it. This is almost a total shot in the dark.

**Comment by jpburstrom** [comment]
Hi, I've tried the #4105 build on a fairly complex ui with several windows, and seems to work just fine on my end. No warning sounds, keyDownActions and globalKeyDownActions work as expected.

**Comment by Xon77** [comment]
the same as @jpburstrom .
I've tried the #4105  build : keyDownActions work fine with no warning sounds.

**Comment by brianlheim** [comment]
Merged #4105. Thanks for testing @jpburstrom @Xon77 !
